src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -350,6 +354,363 @@ def output_gds(self, value: str) -> None:
+        """Set the path to the output netlist file."""
+        if not (isinstance(value, str)):
+            raise TypeError("output_netlist must be a str")
+        self.attr_setter("_output_netlist", value)
+
+    @property
+    def power_nets(self) -> List[str]:
+        """
+        Get the list of power.
+
+        :return: The list of power.
+        """
+        try:
+            return self.attr_getter("_power_nets", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the list of power yet")
+
+    @power_nets.setter
+    def power_nets(self, value: List[str]) -> None:
+        """Set the list of power."""
+        if not (isinstance(value, List)):
+            raise TypeError("power_nets must be a List[str]")
+        self.attr_setter("_power_nets", value)
+
+
+    @property
+    def ground_nets(self) -> List[str]:
+        """
+        Get the list of ground nets in the design.
+
+        :return: The list of ground nets in the design.
+        """
+        try:
+            return self.attr_getter("_ground_nets", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the list of ground nets in the design yet")
+
+    @ground_nets.setter
+    def ground_nets(self, value: List[str]) -> None:
+        """Set the list of ground nets in the design."""
+        if not (isinstance(value, List)):
+            raise TypeError("ground_nets must be a List[str]")
+        self.attr_setter("_ground_nets", value)
+
+
+    @property
+    def hcells_list(self) -> List[str]:
+        """
+        Get the list of hierarchical cells for LVS.
+
+        :return: The list of hierarchical cells for LVS.
+        """
+        try:
+            return self.attr_getter("_hcells_list", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the list of hierarchical cells for LVS yet")
+
+    @hcells_list.setter
+    def hcells_list(self, value: List[str]) -> None:
+        """Set the list of hierarchical cells for LVS."""
+        if not (isinstance(value, List)):
+            raise TypeError("hcells_list must be a List[str]")
+        self.attr_setter("_hcells_list", value)
+
+class HammerSignoffTool(HammerTool):
+    @abstractmethod
+    def fill_outputs(self) -> bool:
+        pass
+
+    ### Inputs ###
+
+    @property
+    def top_module(self) -> str:
+        """
+        Get the top-level module.
+
+        :return: The top-level module.
+        """
+        try:
+            return self.attr_getter("_top_module", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the top-level module yet")
+
+    @top_module.setter
+    def top_module(self, value: str) -> None:
+        """Set the top-level module."""
+        if not (isinstance(value, str)):
+            raise TypeError("top_module must be a str")
+        self.attr_setter("_top_module", value)
+
+    ### Outputs ###
+    @abstractmethod
+    def signoff_results(self) -> int:
```

Comment:
Should we flesh this out into providing more information going forward?

Reply:
I think subclasses of `SignoffTool` should have their own `*_results` methods that can provide more specific details of the issues raised, but I am not strongly attached to that position. Originally this was a `bool` for pass/fail, but I figured an `int` was slightly more descriptive.

Reply:
I think this is fine, but can we document this reasoning in the code somewhere?

Reply:
Done